src/components/item/sharing/ItemSharingTab.tsx

```diff
@@ -33,6 +34,27 @@ type Props = {
   memberships?: ItemMembership[];
 };
 
+interface PermissionMap {
+  [key: string]: ItemMembership;
+}
+
+const reduceMembershipsForSameMember = (
+  memberships: ItemMembership[],
+): ItemMembership[] => {
+  const permissionMap = memberships.reduce((acc: PermissionMap, curr) => {
```

Comment:
FYI, this form also works, both are equivalent:
```suggestion
  const permissionMap = memberships.reduce<PermissionMap>((acc, curr) => {
```